pkg/controller/factory/factory.go

```diff
@@ -34,6 +41,23 @@ type namespaceInformer struct {
 	namespaces sets.String
 }
 
+// PostStartHook specify a function that will run after controller is started.
+// The context is cancelled when the controller is asked to shutdown and the post start hook should terminate as well.
+// The syncContext allow access to controller queue and event recorder.
+type PostStartHook func(ctx context.Context, syncContext SyncContext) error
+
+// ObjectQueueKeyFunc is used to make a string work queue key out of the runtime object that is passed to it.
+// This can extract the "namespace/name" if you need to or just return "key" if you building controller that only use string
+// triggers.
+type ObjectQueueKeyFunc func(runtime.Object) string
+
+// StringQueueKeyFunc is used to make a string work queue key based on string input.
+type StringQueueKeyFunc func(string) string
```

Comment:
is this still used? I hope not. This signature confuses me.

Reply:
not used